website/docs/docs/guides/debugging-schema-names.md

```diff
@@ -0,0 +1,98 @@
+{% macro generate_schema_name(custom_schema_name, node) -%}
+    {{ generate_schema_name_for_env(custom_schema_name, node) }}
+{%- endmacro %}
+```
+Your project is switching out the `generate_schema_name` macro for another macro, `generate_schema_name_for_env`. Similar to the above example, this is a macro which is defined in dbt's global project, [here](https://github.com/fishtown-analytics/dbt/blob/dev/kiyoshi-kuromiya/core/dbt/include/global_project/macros/etc/get_custom_schema.sql#L43-L56).
+
+```sql
+{% macro generate_schema_name_for_env(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if target.name == 'prod' and custom_schema_name is not none -%}
+
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+```
+#### I have a `generate_schema_name` macro with custom logic
+
+If this is the case — it might be a great idea to reach out to the person who added this macro to your project, as they will have context here — you can use [GitHub's blame feature](https://docs.github.com/en/free-pro-team@latest/github/managing-files-in-a-repository/tracking-changes-in-a-file) to do this.
+
+In all cases take a moment to read through the Jinja to see if you can follow the logic.
+
+
+### 2. Confirm your `schema` config
+Check if you are using the [`schema` config](resource-configs/schema) in your model, either via a `{{ config() }}` block, or from `dbt_project.yml`. This gets passed through to the macro as the `custom_schema_name` argument.
+
+
+### 3. Confirm your target values
+Most `generate_schema_name` macros incorporate logic from the [`target` variable](target), in particular `target.schema` and `target.name`. Use the docs [here](target) to help you find the values of each key in this dictionary.
+
+
+### 4. Put the two together
+
+Now, re-read through the logic of your `generate_schema_name` macro, and mentally plug in your `customer_schema_name` and `target` values.
+
+You should find that the schema dbt is constructing for your model matches the output of your `generate_schema_name` macro.
+
+:::info
+Note that snapshots do not follow this behavior, check out the docs on [target_schema](resource-configs/target_schema) instead.
+:::
+
+### 4. Adjust as necessary
+
+Now that you understand how a model's schema is being generated, you can adjust as necessary:
+- You can adjust the logic in your `generate_schema_name` macro (or add this macro to your project if you don't yet have one and adjust from there)
+- You can also adjust your `target` details (for example, changing the name of a target)
+
+If you change the logic in `generate_schema_name`, it's important that you consider whether two users will end up writing to the same schema when in development — we thought ahead for you by implementing name-spaced logic in the the default `generate_schema_name` macro (but acknowledge that we might have been a little too clever here 😅).
```

Comment:
I'm wonder what this parenthetical at the end actually refers to. I always love the personality but it's too vague for me to parse at the moment. If you want to keep the conversational aspect--which I encourage--explain more clearly how we might been a bit too clever. (e.g. but we also acknowledge we may have been a little too clever trying to keep things namespaced). 

I just felt when reading this that there is a conversation about development that I'm only passingly familiar with.